src/systems/fem_system.C

```diff
@@ -74,7 +75,7 @@ void assemble_unconstrained_element_system(const FEMSystem & _sys,
 
   // We need jacobians to do heterogeneous residual constraints
   const bool need_jacobian =
-    _get_jacobian || _constrain_heterogeneously;
+    (_get_jacobian || _constrain_heterogeneously) && (!_no_constraints);
```

Comment:
Yes, in fact, the _no_constraints option isnt needed at all in this function. Fixing this.